test/e2e/azure_test.go

```diff
@@ -556,6 +556,71 @@ var _ = Describe("Workload cluster creation", func() {
 		})
 	})
 
+	// ci-e2e.sh and Prow CI skip this test by default. To include this test, set `GINKGO_SKIP=""`.
+	// This spec expects a user-assigned identity named "cloud-provider-user-identity" in a "capz-ci"
+	// resource group. Override these defaults by setting the USER_IDENTITY and CI_RG environment variables.
+	Context("Creating a cluster that uses the external cloud provider and machinepools [OPTIONAL]", func() {
+		It("with 1 control plane node and 1 machinepool", func() {
+			By("using user-assigned identity")
+			clusterName = getClusterName(clusterNamePrefix, "flex")
+			clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+				ClusterProxy: bootstrapClusterProxy,
+				ConfigCluster: clusterctl.ConfigClusterInput{
+					LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
+					ClusterctlConfigPath:     clusterctlConfigPath,
+					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+					Flavor:                   "external-cloud-provider-machinepool",
```

Comment:
Would it be better to name this new flavor "external-cloud-provider-vmss-flex" to be clear about the fact that we're testing an Azure scenario (flex) vs a CAPI scenario (machinepool)?

Reply:
That would be more on target for this PR. As I commented elsewhere, I intended this template to test the matrix of Windows|Linux * Uniform|Flexible, but started small. I'll rename it and we can rename it again when its functionality changes.

Reply:
Makes sense to me, thanks!